### PR TITLE
Format for clustering CSC rechits with jet algorithms

### PR DESCRIPTION
**Pull request by kakwok**
#### PR description:

This PR adds a new class `CSCJetCandidate` that builds from `CSCRecHit2D` and conform to `Reco::Candidate` format. 
The new collection can be used for running jet algorithms on CSC rechits. 
This is an essential ingredient for an HLT path for LLP in Run 3. 

[Presentation at Reco meeting, Oct 1](https://indico.cern.ch/event/1080450/contributions/4548741/attachments/2320995/3952292/MuonShower_Oct1_Martin.pdf
)

#### PR validation:
```
runTheMatrix.py -l limited -i all --ibeos
31 0 0 0 0 0 0 0 0 tests passed, 8 31 0 0 0 0 0 0 0 failed
```

### CODE REVIEW

File: DataFormats/CSCRecHit/src/CSCJetCandidate.cc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                                       float tWire,
+                                       int quality,
+                                       int chamber,
+                                       int station,
+                                       int nStrips,
+                                       int hitWire,
+                                       int wgroupsBX,
+                                       int nWireGroups)
+    : LeafCandidate(0, LorentzVector(math::PtEtaPhiMLorentzVector(1.0, eta, phi, 0)), Point(0, 0, 0)) {
+  x_ = x;
```
Comment (slava77) [inline DataFormats/CSCRecHit/src/CSCJetCandidate.cc:24]:
```suggestion
    : LeafCandidate(0, LorentzVector(math::PtEtaPhiMLorentzVector(1.0, eta, phi, 0)), Point(0, 0, 0)), x_(x) {
```
initializer list (before `{` is preferred to the use of the constructor body)

[the comment applies to the rest of the members]

**Review by slava77 — COMMENTED** [review]
this is just a quick set of comments on the style

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by VinInn** [comment]
is all this really needed?
  would not be better to directly call FastJet and avoid to create new data objects and copy back and forth info?

**Comment by rappoccio** [comment]
Hi @VinInn we've been doing this at the HLT level since 2009, it's no different from every other jet collection at HLT. 

@ptcox this is for an EXO-based trigger from @kakwok for long-lived jets.

**Comment by kakwok** [comment]
Hi @ptcox , yes, it's a consumer of CSC rechits. We can move the code to `DataFormats/MuonReco` as suggested by @slava77 

Hi @rappoccio, if I understand correctly, @VinInn 's proposal is to write a new producer directly calling `fastjet`, which plays the role of the `fastjetJetProducer` to cluster on csc rechit directly. Would that be possible?

**Comment by dpiparo** [comment]
Isn't @VinInn perhaps trying to hint to the costs implied by the approach?

**Comment by VinInn** [comment]
> On 8 Oct, 2021, at 2:19 PM, rappoccio ***@***.***> wrote:
> 
> We still have to copy rechits to a new DataFormat just like we do now

Why?
 One needs just to populate locally the FastJet input vector

v.

**Comment by rappoccio** [comment]
IMO, "a specific CSC cluster with a RefVector" is a new dataformat that should be encapsulated concisely. 

Anyway, you all have the last word. I think it's not going to be faster in the end, it will duplicate code, and will needlessly add to the long-term maintenance and validation costs. My 0.02 CHF. I'll shut up now :)

**Comment by perrotta** [comment]
@kakwok , since the PR development is going on in #35836, I close this one to clean up the queue.
Please reopen if I misunderstood your intentions

**Pull request closed without merging**